File: ios/QuerosermB/Data/Remote/DTO/ExchangeAssetsDTO.swift
import Foundation

// MARK: - Exchange Assets Response
struct ExchangeAssetsResponse: Codable {
    let data: [ExchangeAssetItem]
}

struct ExchangeAssetItem: Codable {
    let currency: AssetCurrencyData
    let balance: Double?
    let walletAddress: String?

    enum CodingKeys: String, CodingKey {
        case currency
        case balance
        case walletAddress = "wallet_address"
    }
}

struct AssetCurrencyData: Codable {
    let id: Int
    let name: String
    let symbol: String
    let priceUsd: Double?

    enum CodingKeys: String, CodingKey {
        case id = "crypto_id"
        case name, symbol
        case priceUsd = "price_usd"
    }
}

// MARK: - Mapping → Domain
extension ExchangeAssetItem {
    func toDomain() -> Currency {
        Currency(
            id: currency.id,
            name: currency.name,
            symbol: currency.symbol,
            priceUSD: currency.priceUsd,
            balance: balance
        )
    }
}

File: android/app/src/main/java/br/com/querosermb/data/remote/dto/ExchangeAssetsDto.kt
package br.com.querosermb.data.remote.dto

import br.com.querosermb.domain.model.Currency
import com.google.gson.annotations.SerializedName

data class ExchangeAssetsResponse(
    val data: List<ExchangeAssetItem>
)

data class ExchangeAssetItem(
    val currency: AssetCurrencyData,
    val balance: Double?,
    @SerializedName("wallet_address") val walletAddress: String?
) {
    fun toDomain(): Currency = Currency(
        id = currency.id,
        name = currency.name,
        symbol = currency.symbol,
        priceUSD = currency.priceUsd,
        balance = balance
    )
}

data class AssetCurrencyData(
    @SerializedName("crypto_id") val id: Int,
    val name: String,
    val symbol: String,
    @SerializedName("price_usd") val priceUsd: Double?
)
